From 119768308ef9f2d7bec27408892249d9107a9e57 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Tommi=20My=C3=B6h=C3=A4nen?= <ext-tommi.1.myohanen@nokia.com>
Date: Sat, 22 Aug 2009 17:09:39 +0300
Subject: [PATCH] camerabin: fix scene mode setting

Camerabin caches photography settings, but it didn't take into account
that scene mode setting may change other settings as well. So, config
needs to be read back from device after scene mode is set.
---
 gst/camerabin/gstcamerabinphotography.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/gst/camerabin/gstcamerabinphotography.c b/gst/camerabin/gstcamerabinphotography.c
index c568914..31dcb0e 100644
--- a/gst/camerabin/gstcamerabinphotography.c
+++ b/gst/camerabin/gstcamerabinphotography.c
@@ -283,6 +283,10 @@ gst_camerabin_set_scene_mode (GstPhotography * photo, GstSceneMode scene_mode)
   if (PHOTOGRAPHY_IS_OK (camera->src_vid_src)) {
     ret = gst_photography_set_scene_mode (GST_PHOTOGRAPHY (camera->src_vid_src),
         scene_mode);
+    if (ret) {
+      gst_photography_get_config (GST_PHOTOGRAPHY (camera->src_vid_src),
+          &camera->photo_settings);
+    }
   }
   return ret;
 }
